public/js/index.js

```diff
@@ -34,291 +34,314 @@ const clearBtn = document.querySelector(".create-lesson-clear");
 
 // event listener setup
 formElement.addEventListener("submit", function (e) {
-  e.preventDefault();
-  addLesson();
+	e.preventDefault();
+	addLesson();
 });
+
+window.onload = handleClearBtn;
 modalLessonClose.addEventListener("click", handleCloseLessonModal);
 clearBtn.addEventListener("click", handleClear);
 overlay.addEventListener("click", handleCloseLessonModal);
 
 function handleClear(e) {
-  quill.root.innerHTML = "";
-  lessonInput.value = "";
-  clearBtn.setAttribute("hidden", "");
-  submitLessonElement.textContent = "ADD LESSON";
+	quill.root.innerHTML = "";
+	lessonInput.value = "";
+	clearBtn.setAttribute("hidden", "");
+	submitLessonElement.textContent = "ADD LESSON";
 }
 
 function handleViewClick(lesson) {
-  const title = lesson.querySelector(".lesson-card-title").innerText;
-  const content = lesson.querySelector(".lesson-card-content").innerHTML;
-
-  modalLessonTitle.innerText = title;
-  modalLessonContent.innerHTML = content;
-  modalLesson.removeAttribute("hidden");
-  overlay.removeAttribute("hidden");
-  overlay.classList.add("dark");
+	const title = lesson.querySelector(".lesson-card-title").innerText;
+	const content = lesson.querySelector(".lesson-card-content").innerHTML;
+
+	modalLessonTitle.innerText = title;
+	modalLessonContent.innerHTML = content;
+	modalLesson.removeAttribute("hidden");
+	overlay.removeAttribute("hidden");
+	overlay.classList.add("dark");
+}
+
+function handleClearBtn() {
+	formElement.mouseOver = false;
+	formElement.onmouseover = () => {
+		if (!this.mouseOver) {
+			this.mouseOver = true;
+			clearBtn.removeAttribute('hidden')
+		}
+	}
+	formElement.onmouseleave = () => {
+		if (this.mouseOver) {
+			if (lessonInput.value.length === 0) {
+				clearBtn.setAttribute('hidden', '');
+				this.mouseOver = false;
+			}
+			if (quill.root.innerHTML !== '<p><br></p>' && quill.root.innerHTML.length >= 8) {
+				clearBtn.removeAttribute('hidden');
+				this.mouseOver = false;
+			}
+		}
+	}
 }
```

Comment:
This, is the implemented function to handle the Clear button. I added it in the event listeners setup.